src/NuGet.Core/NuGet.Versioning/FloatRange.cs

```diff
@@ -150,23 +178,77 @@ public static bool TryParse(string versionString, out FloatRange range)
 
             if (versionString != null)
             {
-                var starPos = versionString.IndexOf('*');
-
-                var actualVersion = versionString;
+                var firstStarPosition = versionString.IndexOf('*');
+                var lastStarPosition = versionString.LastIndexOf('*');
                 string releasePrefix = null;
 
                 if (versionString.Length == 1
-                    && starPos == 0)
+                    && firstStarPosition == 0)
                 {
                     range = new FloatRange(NuGetVersionFloatBehavior.Major, new NuGetVersion(new Version(0, 0)));
                 }
-                // * must appear as the last char in the string. 
+                else if (versionString.Equals("*-*"))
+                {
+                    range = new FloatRange(NuGetVersionFloatBehavior.AbsoluteLatest, new NuGetVersion("0.0.0-0"), releasePrefix: string.Empty);
+                }
+                else if (firstStarPosition != lastStarPosition && lastStarPosition != -1 && versionString.IndexOf('+') == -1)
```

Comment:
Is "&& versionString.IndexOf('+') == -1" needed? How *-*+001 will be resolved?

Reply:
`+` is only valid in a version's metadata, which not part of the version identity.
As such we do not allow it in floating versions,